Maple2.Server.Game/Manager/Config/BuffManager.cs

```diff
@@ -143,6 +168,128 @@ public class BuffManager : IUpdatable {
         Reflect = record;
     }
 
+
+    private void SetInvoke(Buff buff) {
+        if (buff.Metadata.InvokeEffect == null) {
+            return;
+        }
+
+        for (int i = 0; i < buff.Metadata.InvokeEffect.Types.Length; i++) {
+            var record = new InvokeRecord(buff.Id, buff.Metadata.InvokeEffect) {
+                Value = buff.Metadata.InvokeEffect.Values[i],
+                Rate = buff.Metadata.InvokeEffect.Rates[i],
+            };
+
+            // if exists, replace the record to support differentiating buff levels.
+            if (Invokes.TryGetValue(buff.Metadata.InvokeEffect.Types[i], out IDictionary<int, InvokeRecord>? nestedInvokeDic)) {
+                Invokes.RemoveAll(buff.Id);
+
+                if (!nestedInvokeDic.TryAdd(buff.Id, record)) {
+                    logger.Error("Could not add invoke record from {Id} to {Object}", buff.Id, actor.ObjectId);
+                }
+                continue;
+            }
+
+            if (!Invokes.TryAdd(buff.Metadata.InvokeEffect.Types[i], new ConcurrentDictionary<int, InvokeRecord> {
```

Comment:
Curious if `TryAdd` is expected to fail often here. Reason being we are instantiating a new dictionary only for it to fail. If it's expected to fail often, it would be better to defer the creation of the dictionary.

Reply:
No it rarely should ever fail. I pushed a change to just use Add instead